src/include/miopen/conv/problem_description.hpp

```diff
@@ -161,13 +161,13 @@ struct ProblemDescription
     std::size_t GetInStrideD() const { return GetD5(GetSpatialDims(), in.GetStrides()); }
     std::size_t GetInStrideH() const { return GetH5(GetSpatialDims(), in.GetStrides()); }
     std::size_t GetInStrideW() const { return GetW5(GetSpatialDims(), in.GetStrides()); }
-    std::string GetInLayout() const { return "NCHW"; }
+    std::string GetInLayout() const { return in.GetLayout(); }
     std::size_t GetInElementSize() const { return GetTypeSize(GetInDataType()); }
 
     std::size_t GetInSize() const
     {
         // clang-format off
-        return (GetInLayout() == "NCHW")
+        return (GetInLayout() == "nchw")
```

Comment:
Why this change?

Reply:
The is related with the previous change: since input layout get defaulted to the lower case "nchw", the comparison is using "nchw". This can be addressed by, say, we formalized the layout string to be all lower or upper case.

Though I don't think I fully understand why it was originally written to compare with the upper cased "NCHW". Since `GetInLayout()` already get hard-coded to `NCHW` what's the point of the comparison here? Is it aiming for some future non-packed data?

Reply:
> since input layout get defaulted to the lower case "nchw"

This change is undesired, please see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/343#discussion_r456054013.

> Though I don't think I fully understand why it was originally written to compare with the upper cased "NCHW".

I am far from even moderate understanding. All this is legacy stuff, survived 4 years of refactoring and other changes. Originated from https://github.com/AMDComputeLibraries/MLOpen/pull/5/files#r456075197 4 years ago. It looks like dead code now.

And it will become a bug as soon as `GetInLayout()` will begin to return "NCDHW" for 3D tensors.

[Recommendation] Get rid of this in a separate PR.

Reply:
Sounds good, will return here, but file a separate PR to return `GetInBatchSize() * GetInChannels() * GetInDepth() * GetInHeight() * GetInWidth() * GetInElementSize()` directly.

Reply:
My previous recommendation (getting rid of ternary) also applies to GetOutSize, GetWeightsSize

Reply:
> Sounds good, will revert here, but file a separate PR 

I recommend keeping this unchanged here, to avoid merge conflicts. Up to you of course.